src/main/scala/it/unibo/authsim/library/dsl/cryptography/algorithm/symmetric/AES.scala

```diff
@@ -0,0 +1,86 @@
+package it.unibo.authsim.library.dsl.cryptography.algorithm.symmetric
+
+import it.unibo.authsim.library.dsl.cryptography.algorithm.SymmetricEncryptionAlgorithm
+
+/**
+ * Trait for AES algorithm
+ */
+trait AES extends SymmetricEncryptionAlgorithm:
+  /**
+   * Setter for the length of the key to be used in the encryption operation.
+   *
+   * @param newKeyLength          new key length
+   */
+  def keyLength_(newKeyLength:Int):Unit
+
+/**
+ * Companion object of the AES trait
+ */
+object AES:
+  import it.unibo.authsim.library.dsl.cryptography.encrypter.asymmetric.key.KeysGenerator
+
+  /**
+   * Apply method for the object
+   * @return an istance of the AES case class
+   */
+  def apply()= new BasicAES()
+
+  /**
+   * Class representing a basic implementation of the AES algorithm
+   */
+  case class BasicAES() extends AES:
+
+    type Salt = String
+    /**
+     * Private variable representing the length of the key used during the cryptographic operation
+     */
+    private var _length : Int = 16 
+    
+    /**
+     * Private variable representing the algorithm name
+     */
+    val _name : String ="AES"
+
+    /**
+     * Private variable representing the length of the key algorithm supports
+     */
+    private var keySet = Set(16, 24, 32)
```

Comment:
This variable seems to not change in this class, it should be made `val`:
```suggestion
    private val keySet = Set(16, 24, 32)
```